### PR TITLE
Introduce a UBI9 builder

### PR DESCRIPTION
**Pull request by lfrancke**
# Description

*Please add a description here.*

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
- [ ] Changes are OpenShift compatible
- [ ] All added packages (via microdnf or otherwise) have a comment on why they are added
- [ ] Things not downloaded from Red Hat repositories should be mirrored in the Stackable repository and downloaded from there
- [ ] All packages should have (if available) signatures/hashes verified
- [ ] Does your change affect an SBOM? Make sure to update all SBOMs
```

### CODE REVIEW

File: ubi9-rust-builder/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# 9.3-1552 as of 2024-03-03
+# https://catalog.redhat.com/software/containers/ubi9/ubi-minimal/615bd9b4075b022acc111bf5?architecture=amd64&image=65a828e3cda4984705d45d26&container-tabs=gti
+# Manifest list digest because of multi architecture builds ( https://www.redhat.com/architect/pull-container-image#:~:text=A%20manifest%20list%20exists%20to,system%20on%20a%20specific%20architecture )
+FROM registry.access.redhat.com/ubi9/ubi-minimal@sha256:582e18f13291d7c686ec4e6e92d20b24c62ae0fc72767c46f30a69b1a6198055 AS builder
```
Comment (Maleware) [inline ubi9-rust-builder/Dockerfile:4]: This should be the manifest list from https://catalog.redhat.com/software/containers/ubi9/ubi-minimal/615bd9b4075b022acc111bf5?architecture=amd64&image=65e0932f034203e025b55a92&container-tabs=gti ( ubi 9.3 | latest c)

So `registry.access.redhat.com/ubi9/ubi-minimal@sha256:bc552efb4966aaa44b02532be3168ac1ff18e2af299d0fe89502a1d9fabafbc5`

Comment (lfrancke) [reply ubi9-rust-builder/Dockerfile]: Thanks! I hope I fixed it. Can you check if this looks better?

File: ubi9-rust-builder/copy_artifacts.sh
Comment (soenkeliebau) [inline ubi9-rust-builder/copy_artifacts.sh:1]: This seems to be an exact duplicate, can we maybe share this file?

File: ubi9-rust-builder/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  && unzip protoc.zip \
+  && rm protoc.zip
+ENV PROTOC=/opt/protoc/bin/protoc
+WORKDIR /
+
+# IMPORTANT
+# If you change the toolchain version here, make sure to also change the "rust_version"
+# property in operator-templating/repositories.yaml
+# hadolint ignore=SC1091
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --default-toolchain 1.75.0 \
```
Comment (soenkeliebau) [inline ubi9-rust-builder/Dockerfile:49]: UBI8 uses 1.77.0 is this intentional?

File: ubi9-rust-builder/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    libkadm5 \
+    make \
+    openssl-devel \
+    pkg-config \
+    systemd-devel \
+    unzip \
+  && rm -rf /var/cache/yum
+
+WORKDIR /opt/protoc
+RUN PROTOC_VERSION=21.5 \
```
Comment (soenkeliebau) [inline ubi9-rust-builder/Dockerfile:37]: This is different from the version in the ubi8 Dockerfile: 26.1 ?

**Review by soenkeliebau — COMMENTED** [review]
just three tiny things I noticed when looking over the pr

File: ubi9-rust-builder/Dockerfile (hunk 2 above)
Comment (lfrancke) [reply ubi9-rust-builder/Dockerfile:49]: Ah, no. I just forgot to update this. Thanks for the catch.

File: ubi9-rust-builder/Dockerfile (hunk 3 above)
Comment (lfrancke) [reply ubi9-rust-builder/Dockerfile:37]: Same here, thanks for the catch! This was recently updated in ubi8 and i forgot to update it here

File: ubi9-rust-builder/copy_artifacts.sh
Comment (lfrancke) [reply ubi9-rust-builder/copy_artifacts.sh:1]: Done

**Review by maltesander — APPROVED** [review]
LGTM!

**Pull request merged**